src/libraries/Pool.sol

```diff
@@ -384,7 +394,7 @@ library Pool {
             // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
             if (cache.protocolFee > 0) {
                 // calculate the amount of the fee that should go to the protocol
-                uint256 delta = step.feeAmount * cache.protocolFee / ProtocolFeeLibrary.BIPS_DENOMINATOR;
+                uint256 delta = step.feeAmount * cache.protocolFee / (swapFee + cache.protocolFee);
```

Comment:
i feel like this can be unchecked? i guess it wasnt unchecked before either...
but `max(uint256) * max(protocolFee)` is not going to overflow
and `swapFee + protocolFee` is also not going to overflow?

Reply:
`swapFee + protocolFee` may overflow unless written as `uint256(swapFee) + protocolFee`.